### PR TITLE
EREGCSC-2113 -- Reflect edited statute dates in frontend

### PR DESCRIPTION
**Pull request by PhilR8**
Resolves [EREGCSC-2113](https://jiraent.cms.gov/browse/EREGCSC-2113)

**Description:**

Pulls dates from site configuration in admin panel into front end for Statute Table column header labels.

**This pull request changes:**

- Adds statute table site configuration labels and dates into Statutes template using [best practice](https://adamj.eu/tech/2022/10/06/how-to-safely-pass-data-to-javascript-in-a-django-template/)
- Only shows dates if both label and date exist, and if date is a valid date
- Updates snapshot tests and adds unit front end unit test coverage

**Steps to manually verify this change:**

1. Go into [admin panel for experimental deployment](https://1ghm0p11p3.execute-api.us-east-1.amazonaws.com/dev929/admin/) > Regulations > Site Configuration
2. Edit Date Type and Date
3. Try out a bunch of options.  If Date Type and Date are both valid, the date subtitle should appear in the column header.  If one or both of the Date Type and Date are empty strings or invalid, the date subtitle will not appear.

### CODE REVIEW

File: solution/backend/regulations/templates/regulations/statute.html [hunk 1]
```diff
@@ -18,6 +18,7 @@
 {% endblock %}
 
 {% block body %}
+{{ site_config|json_script:"site_config" }}
```
Comment (PhilR8) [inline solution/backend/regulations/templates/regulations/statute.html:21]: Passing in complex values using the Django best practice discussed here: https://adamj.eu/tech/2022/10/06/how-to-safely-pass-data-to-javascript-in-a-django-template/

File: solution/backend/regulations/views/statute.py [hunk 2]
```diff
@@ -1,15 +1,39 @@
...
+            },
+            'statute_compilation': {
+                'type': site_config.statute_compilation_date_type,
+                'date': site_config.statute_compilation_date,
+            },
+            'ssa_gov_compilation': {
+                'type': site_config.ssa_gov_compilation_date_type,
+                'date': site_config.ssa_gov_compilation_date,
+            },
+        }
```
Comment (PhilR8) [inline solution/backend/regulations/views/statute.py:32]: Emulated the pattern used in `reader.py` view to bring in site config values to use in template

File: solution/ui/regulations/eregs-component-lib/src/components/shared-components/Statutes/schemas/tableSchemas.js [hunk 3]
```diff
@@ -1,4 +1,10 @@
-import { houseGovUrl, statuteCompilationUrl, usCodeUrl, ssaGovUrl } from "../utils/urlMethods.js";
+import {
+    houseGovUrl,
+    statuteCompilationUrl,
+    usCodeUrl,
```
Comment (thwalker6) [inline solution/ui/regulations/eregs-component-lib/src/components/shared-components/Statutes/schemas/tableSchemas.js:4]: Doesnt need to be done here but should we maybe have our imports sorted alphabetically. Its currently what we do with Django.  Granted we have Ruff automatically order them for us so its much easier to manage.

**Review by thwalker6 — APPROVED** [review]
it looks good.  Just left notes here for discussion as a team.  LGTM

File: solution/ui/regulations/eregs-component-lib/src/components/shared-components/Statutes/schemas/tableSchemas.js (hunk 3 above)
Comment (PhilR8) [reply solution/ui/regulations/eregs-component-lib/src/components/shared-components/Statutes/schemas/tableSchemas.js:5]: alphabetized!

**Pull request merged**